zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java

```diff
@@ -179,12 +274,210 @@ public static Collection getRealmsList() {
+            currentUser = (String) attrs.get(userAttribute).get();
+          }
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("CurrentUser: " + currentUser);
+          }
+          userList.add(currentUser.trim());
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.error("Error retrieving User list from Ldap Realm", e);
+    }
+    return userList;
+  }
+
+  /***
+   * Get user roles from shiro.ini for Zeppelin LdapRealm.
+   *
+   * @param r
+   * @return
+   */
+  private static List<String> getRolesList(LdapRealm r) {
+    List<String> roleList = new ArrayList<>();
+    Map<String, String> roles = r.getListRoles();
+    if (roles != null) {
+      Iterator it = roles.entrySet().iterator();
+      while (it.hasNext()) {
+        Map.Entry pair = (Map.Entry) it.next();
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.debug("RoleKeyValue: " + pair.getKey() +
+              " = " + pair.getValue());
+        }
+        roleList.add((String) pair.getKey());
+      }
+    }
+    return roleList;
+  }
+
+  private static List<String> getUserList(ActiveDirectoryGroupRealm r, String searchText,
+                                   int numUsersToFetch) {
+    List<String> userList = new ArrayList<>();
+    try {
+      LdapContext ctx = r.getLdapContextFactory().getSystemLdapContext();
+      userList = r.searchForUserName(searchText, ctx, numUsersToFetch);
+    } catch (Exception e) {
+      LOGGER.error("Error retrieving User list from ActiveDirectory Realm", e);
+    }
+    return userList;
+  }
+
+  /**
+   * Function to extract users from JDBCs.
+   */
+  private static List<String> getUserList(JdbcRealm obj) {
+    List<String> userlist = new ArrayList<>();
+    Connection con = null;
+    PreparedStatement ps = null;
+    ResultSet rs = null;
+    DataSource dataSource = null;
+    String authQuery = "";
+    String retval[];
+    String tablename = "";
+    String username = "";
+    String userquery;
+    try {
+      dataSource = (DataSource) FieldUtils.readField(obj, "dataSource", true);
+      authQuery = (String) FieldUtils.readField(obj, "authenticationQuery", true);
+      LOGGER.info(authQuery);
+      String authQueryLowerCase = authQuery.toLowerCase();
+      retval = authQueryLowerCase.split("from", 2);
+      if (retval.length >= 2) {
+        retval = retval[1].split("with|where", 2);
+        tablename = retval[0];
+        retval = retval[1].split("where", 2);
+        if (retval.length >= 2) {
+          retval = retval[1].split("=", 2);
+        } else {
+          retval = retval[0].split("=", 2);
+        }
+        username = retval[0];
+      }
+
+      if (StringUtils.isBlank(username) || StringUtils.isBlank(tablename)) {
+        return userlist;
+      }
+
+      userquery = String.format("SELECT %s FROM %s", username, tablename);
+    } catch (IllegalAccessException e) {
+      LOGGER.error("Error while accessing dataSource for JDBC Realm", e);
+      return null;
```

Comment:
seems like we should return userlist (would be empty) here?
or L475 should also return null?

might just need to be consistent

Reply:
will do it